packages/forms/examples/file-upload.md

```diff
@@ -0,0 +1,161 @@
+  border-color: ${p => getColor('neutralHue', 300, p.theme)};
+  border-radius: ${p => p.theme.borderRadii.md};
+  padding: ${p => p.theme.space.xxs} ${p => p.theme.space.xs};
+  display: inline-flex;
+  min-width: 200px;
+  flex-wrap: nowrap;
+`;
+
+const StyledFileName = styled.span`
+  display: flex;
+  align-items: center;
+  flex-grow: 1;
+  padding: 0 ${p => p.theme.space.xs};
+`;
+
+const StyledFileIcon = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const StyledProgress = styled(Progress)`
+  position: absolute;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  margin: 0;
+`;
+
+const File = React.memo(({ name, onRemove }) => {
+  const [uploadProgress, setUploadProgress] = React.useState(0);
+
+  React.useEffect(() => {
+    const randomUploadInterval = Math.random() * (600 - 200) + 600;
+
+    const uploadInterval = setInterval(() => {
+      setUploadProgress(prevProgress => {
+        if (prevProgress >= 100) {
+          clearInterval(uploadInterval);
+          return 100;
+        }
+
+        return prevProgress + 10;
+      });
+    }, randomUploadInterval);
+
+    return () => {
+      clearInterval(uploadInterval);
+    };
+  }, []);
+
+  return (
+    <StyledFile aria-label="File">
+      <StyledFileIcon>
+        <FileImageStroke />
+      </StyledFileIcon>
+      <StyledFileName aria-label="File name">{name}</StyledFileName>
+      <IconButton size="small" onClick={onRemove} aria-label="Remove file">
+        <CloseStroke />
+      </IconButton>
+      {uploadProgress < 100 && (
+        <StyledProgress value={uploadProgress} aria-label="Upload progress" />
+      )}
+    </StyledFile>
+  );
+});
+
+const Example = () => {
+  const [isCompact, setIsCompact] = React.useState(false);
+  const [files, setFiles] = React.useState(['squash.jpg', 'soybean.pdf']);
+
+  const onDrop = React.useCallback(
+    acceptedFiles => {
+      if (acceptedFiles && acceptedFiles.length > 0) {
+        const fileNames = acceptedFiles.map(file => file.name);
+
+        setFiles([...files, ...fileNames]);
+      }
+    },
+    [files]
+  );
+
+  const removeFile = React.useCallback(
+    fileIndex => {
+      setFiles(files.filter((file, i) => i !== fileIndex));
+    },
+    [files]
+  );
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });
+
+  return (
+    <>
+      <Well isRecessed className="u-mb-sm">
+        <Field>
+          <Toggle checked={isCompact} onChange={event => setIsCompact(event.target.checked)}>
+            <Label>
+              <Code>isCompact</Code>
+            </Label>
+          </Toggle>
+        </Field>
+      </Well>
+      <Field>
+        <Label>File upload</Label>
+        <Hint>Works with react-dropzone</Hint>
+        <FileUpload {...getRootProps()} isDragging={isDragActive} isCompact={isCompact}>
+          {isDragActive ? (
+            <span>Drop files here</span>
+          ) : (
+            <span>Drag files here or click to upload</span>
```

Comment:
Recommend replacing these `<span>` with `<Ellipsis>` as we'd prefer to not wrap at extreme sizes.

Reply:
With this being instructional text, I don't believe this should be using an ellipsis. Similar to label/hint/message this should wrap to display the full content. I don't think there is anything positioned relative to this that would require that styling.

Here is the design mocks which would almost definitely be truncated if it was localized with this styling due to the string length:
![image](https://user-images.githubusercontent.com/4030377/92160600-222e2f80-ede4-11ea-9501-9049e022b1c8.png)